calcom/cal.com
# feat: min max chars for long text

## What does this PR do?

Part of #15577 

This PR adds an optional minimum and maximum character limit to long text question in the booking form.

Loom Video: https://www.loom.com/share/735a6768a92847cabefd1fe2187c499d?sid=2d606562-c6e6-4bd3-92c0-b1fcdab14218

- Fixes #15577 
- Fixes [CAL-3973](https://linear.app/calcom/issue/CAL-3973/booking-questions-for-long-text-add-min-and-max-characters-both)

## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?



- Are there environment variables that should be set?
 No

- What are the minimal test data to have?
 NA

- What is expected (happy path) to have (input and output)?
Click on Event Types in the sidebar.
Click on an event.
Inside the event, click on the Advanced tab on the left sidebar.
Add / edit a question.
Set type to Long text
Set the min &/or max characters.
Save the question.
Save the form.
Click on the Preview button in the event header.
Select a date on the calendar.
While filling out the form, click save button with:
a. Min &/or max character limit unsatisfied -> should show red text with error.
b. All the conditions satisfied -> Should save successfully.

- Any other important info that could help to test that PR

/claim https://github.com/calcom/cal.com/issues/15577
